### PR TITLE
fix(tui): do not invoke loop recursively for pad()

### LINKED ISSUE #21610: nvim crashes when `visualbell` is enabled
### Neovim version (nvim -v)

v0.9.0-dev-590+g6ba34e21f

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Linux

### Terminal name/version

wezterm 20221221-163401-72da5d16

### $TERM environment variable

wezterm

### Installation

Built from git head today

### How to reproduce the issue

nvim aborts instantly when visualbell is on and the bell rings (e.g. by moving the cursor out of bounds).

Here's a stacktrace from a RelWithDebugInfo build:

```
Program received signal SIGABRT, Aborted.
__pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
44            return INTERNAL_SYSCALL_ERROR_P (ret) ? INTERNAL_SYSCALL_ERRNO (ret) : 0;
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007ffff7d53ee3 in __pthread_kill_internal (signo=6, threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007ffff7d03aa6 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007ffff7ced7fc in __GI_abort () at abort.c:79
#4  0x00000000004f022c in loop_uv_run (loop=<optimized out>, ms=<optimized out>, once=false) at /home/bernie/src/vim/neovim/src/nvim/event/loop.c:49
#5  0x0000000000659c75 in pad (ctx=<optimized out>, delay=<optimized out>, scale=<optimized out>, force=<optimized out>) at /home/bernie/src/vim/neovim/src/nvim/tui/tui.c:1703
#6  0x00000000003afed6 in unibi_format ()
#7  0x000000000065671a in unibi_out (ui=0x70b280, unibi_index=unibi_index@entry=131) at /home/bernie/src/vim/neovim/src/nvim/tui/tui.c:1651
#8  0x0000000000656759 in tui_visual_bell (ui=<optimized out>) at /home/bernie/src/vim/neovim/src/nvim/tui/tui.c:1309
#9  0x000000000065a58d in ui_call_visual_bell () at /home/bernie/src/vim/neovim/build/src/nvim/auto/ui_events_call.generated.h:43
#10 0x000000000065a5bb in ui_client_event_visual_bell (args={size = 0, capacity = 0, items = 0x759c08}) at /home/bernie/src/vim/neovim/build/src/nvim/auto/ui_events_client.generated.h:59
#11 0x00000000005942fe in parse_msgpack (channel=channel@entry=0x6e7430) at /home/bernie/src/vim/neovim/src/nvim/msgpack_rpc/channel.c:252
#12 0x0000000000594489 in receive_msgpack (stream=<optimized out>, rbuf=0x6e7c30, c=<optimized out>, data=0x6e7430, eof=<optimized out>)
    at /home/bernie/src/vim/neovim/src/nvim/msgpack_rpc/channel.c:236
#13 0x00000000004f1f98 in read_event (argv=argv@entry=0x7fffffffa050) at /home/bernie/src/vim/neovim/src/nvim/event/rstream.c:192
#14 0x00000000004f20ad in invoke_read_cb (stream=stream@entry=0x6e7638, count=count@entry=117, eof=eof@entry=false) at /home/bernie/src/vim/neovim/src/nvim/event/rstream.c:205
#15 0x00000000004f3386 in read_cb (uvstream=<optimized out>, cnt=117, buf=<optimized out>) at /home/bernie/src/vim/neovim/src/nvim/event/rstream.c:134
#16 0x0000000000395687 in uv__read (stream=stream@entry=0x6e7640) at /home/bernie/src/vim/neovim/.deps/build/src/libuv/src/unix/stream.c:1158
#17 0x0000000000395db8 in uv__stream_io (loop=<optimized out>, w=0x6e76c8, events=1) at /home/bernie/src/vim/neovim/.deps/build/src/libuv/src/unix/stream.c:1226
#18 0x000000000039af73 in uv__io_poll (loop=loop@entry=0x6adb00 <main_loop>, timeout=<optimized out>) at /home/bernie/src/vim/neovim/.deps/build/src/libuv/src/unix/linux-core.c:449
#19 0x000000000038bcd5 in uv_run (loop=loop@entry=0x6adb00 <main_loop>, mode=mode@entry=UV_RUN_ONCE) at /home/bernie/src/vim/neovim/.deps/build/src/libuv/src/unix/core.c:406
#20 0x00000000004f0254 in loop_uv_run (loop=loop@entry=0x6adb00 <main_loop>, ms=-1, once=once@entry=true) at /home/bernie/src/vim/neovim/src/nvim/event/loop.c:65
#21 0x00000000004f0bdc in loop_poll_events (loop=0x6adb00 <main_loop>, ms=<optimized out>) at /home/bernie/src/vim/neovim/src/nvim/event/loop.c:87
#22 0x000000000065ebdf in ui_client_run (remote_ui=<optimized out>) at /home/bernie/src/vim/neovim/src/nvim/ui_client.c:91
#23 0x000000000055fae6 in main (argc=1, argv=0x7fffffffd808) at /home/bernie/src/vim/neovim/src/nvim/main.c:372
```

### Expected behavior

Shouldn't crash.

### Actual behavior

But it crashes.

### PR DESCRIPTION
**Pull request by bfredl**
fixes #21610

### CODE REVIEW

File: src/nvim/tui/tui.c [hunk 1]
```diff
@@ -1626,7 +1626,7 @@ static void pad(void *ctx, size_t delay, int scale FUNC_ATTR_UNUSED, int force)
   }
 
   flush_buf(tui);
-  loop_uv_run(tui->loop, (int64_t)(delay / 10), false);
+  usleep((useconds_t)(delay*100));
```
Comment (zeertzjq) [inline src/nvim/tui/tui.c:1629]: `uv_sleep()`?

File: test/functional/terminal/tui_spec.lua [hunk 2]
```diff
@@ -1469,6 +1469,33 @@ describe('TUI', function()
                                                                                  |
       {3:-- TERMINAL --}                                                             |]]))
   end)
+
+  it('visual bell (padding) does not crash #21610', function()
+    feed_data ':set visualbell\n'
```
Comment (zeertzjq) [inline test/functional/terminal/tui_spec.lua:1474]: I think this test also need setting `belloff=`

Comment (bfredl) [reply test/functional/terminal/tui_spec.lua:1474]: I don't think it does. it crashes on master.

Comment (zeertzjq) [reply test/functional/terminal/tui_spec.lua:1474]: Oh, forgot that `nvim_argv` adds this by default.

**Pull request merged**